subsys/net/ip/l2/arp.c

```diff
@@ -222,6 +222,10 @@ struct net_pkt *net_arp_prepare(struct net_pkt *pkt)
 	if (!net_if_ipv4_addr_mask_cmp(net_pkt_iface(pkt),
 				       &NET_IPV4_HDR(pkt)->dst)) {
 		addr = &net_pkt_iface(pkt)->ipv4.gw;
+		if (IS_ENABLED(CONFIG_NET_LOG)
+		    && net_is_ipv4_addr_unspecified(addr)) {
+			NET_ERR("Gateway not set for if %p", net_pkt_iface(pkt));
```

Comment:
"for if"? you mean interface?

Reply:
For network interfaces, we usually have "iface" in the code and in prints, so I suggest that too here.

Reply:
> "for if"? you mean interface?

Well, some people raised an issue that enabling error logging by default bloats binary size, so I'm making changes with that in mind. There're more cryptic messages in the codebase: https://jira.zephyrproject.org/browse/ZEP-1086 (It's closed not because it's fixed, by because I "gave up").

Anyway, changing to "iface" as suggested by @jukkar .